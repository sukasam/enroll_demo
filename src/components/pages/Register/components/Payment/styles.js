import { css } from "@emotion/react";

export default css`
    .paymentSection {
        position: relative;
        border-bottom: 1px solid #ccc;
        padding: 0 0 16px 0;
        margin-top: 32px;

        @media (min-width: 600px) {
            margin-top: 0;
        }
    }

    font-family: Inter, Arial, Helvetica, sans-serif;
    color: #153862;
    border: none !important;

    .label {
        padding: 5px 10px;
    }

    .formTitle {
        font-size: 20px;
        text-align: left;
        font-weight: bold;
    }

    .form {
        display: flex;
        flex-direction: column;
        margin: 0 auto;
        padding: 0 20px;
        border-radius: 8px;
        background-color: white;
        gap: 10px;
    }

    .formRow {
        margin-bottom: 10px;
    }

    .input.invalid {
        border: 1px solid red;
    }

    .button:hover {
        background: #4a75a0;
    }
    .button {
        padding: 10px;
        font-size: 18px;
        color: white;
        background-color: #5a8fc3;
        border: none;
        border-radius: 30px;
        cursor: pointer;
        transition: background-color 0.3s ease;
        max-width: 300px;
    }

    .button.disabled {
        background-color: #cccccc;
        color: #666666;
        cursor: not-allowed;
        border: 1px solid #999999;
        opacity: 0.5;
    }

    .error {
        color: #ff0000;
        font-size: 13px;
        font-weight: 400;
        margin: 0;
        width: 100%;
        min-height: 20px;
        visibility: hidden;
    }
    .error.active {
        visibility: visible;
    }

    .terms-error {
        margin-top: -10px;
    }

    @media (max-width: 768px) {
        .formField.halfWidth {
            width: 100%;
        }

        .formRow {
            flex-direction: column;
        }
    }
`;
