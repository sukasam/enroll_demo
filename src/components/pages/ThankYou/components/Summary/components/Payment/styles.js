import { css } from "@emotion/react";

export default css`
    .payment-row {
        margin-top: 10px;
        display: flex;
        margin-bottom: 10px;
        gap: 8px;
        font-weight: bold;
    }
    .banner {
        display: flex;
        align-items: center;
        justify-content: flex-start;
        background-color: #ebf2f9;
        padding: 10px;
        border-radius: 5px;
        width: 100%;
        margin-top: 10px;
    }
    .banner-header {
        width: 90%;
        margin-left: 10px;
    }
    .banner-body {
        width: 100%;
        margin: 10px 0;
    }
    .banner-body {
        padding: 10px;
        margin: 0 auto;
    }
`;
