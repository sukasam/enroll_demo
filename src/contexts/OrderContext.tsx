import { TransactionMethod } from "Services/Hydra/order/types";

import {
    ReactElement,
    ReactNode,
    createContext,
    useCallback,
    useContext,
    useEffect,
    useMemo,
    useState
} from "react";

import useOrderCalculations from "./hooks/OrderCalculations";
import { useProducts } from "./ProductContext";

import {
    AddressType,
    BankWireInfo,
    CashPaymentInfo,
    EventDetails,
    OrderContextType,
    OrderResult,
    PayPalInfo,
    PaymentMethod,
    ShippingMethodQuoteType
} from "./types/OrderContextTypes";

const defaultOrderContext: OrderContextType = {
    orderResult: null,
    setOrderResult: () => null,
    shippingMethodQuotes: [],
    setShippingMethodQuotes: () => [],
    selectedShippingMethod: null,
    setSelectedShippingMethod: () => null,
    selectedPaymentMethod: null,
    setSelectedPaymentMethod: () => null,
    shipToAddress: null,
    setShipToAddress: () => null,
    signedTerms: false,
    setSignedTerms: () => false,
    userSignature: null,
    setUserSignature: () => null,
    taxId: "",
    setTaxId: () => null,
    secondaryTaxId: "",
    setSecondaryTaxId: () => null,
    paymentMethod: null,
    setPaymentMethod: () => null,
    billingAddress: null,
    setBillingAddress: () => null,
    signatureDateTimeStamp: null,
    setSignatureDateTimeStamp: () => null,
    payPal: null,
    setPayPal: () => null,
    cashPayment: null,
    setCashPayment: () => null,
    bankWire: null,
    setBankWire: () => null,
    isShipToOffice: false,
    setIsShipToOffice: () => false,
    isCreditCardPaymentValid: false,
    setIsCreditCardPaymentValid: () => false,
    selectedWillCallAddress: null,
    setSelectedWillCallAddress: () => null,
    pickUpAddressKey: null,
    setPickUpAddressKey: () => null,
    tppPayload: null,
    setTppPayload: () => null,
    isDigitalOnlyOrder: false,
    feedbackSubmitted: false,
    setFeedbackSubmitted: () => false,
    isEnrollmentCompleted: false,
    setIsEnrollmentCompleted: () => false
};

const OrderContext = createContext<OrderContextType>(defaultOrderContext);
OrderContext.displayName = "Order Context";

export function OrderProvider({
    children,
    testValues
}: {
    children: ReactNode;
    testValues?: Record<string, string>;
}): ReactElement {
    const { shoppingCart } = useProducts();
    const [orderResult, setOrderResult] = useState<OrderResult | null>(
        defaultOrderContext.orderResult
    );
    const [shippingMethodQuotes, setShippingMethodQuotes] = useState<
        ShippingMethodQuoteType[]
    >(defaultOrderContext.shippingMethodQuotes);
    const [selectedShippingMethod, setSelectedShippingMethod] =
        useState<ShippingMethodQuoteType | null>(
            defaultOrderContext.selectedShippingMethod
        );
    const [selectedPaymentMethod, setSelectedPaymentMethod] =
        useState<TransactionMethod | null>(
            defaultOrderContext.selectedPaymentMethod
        );
    const [shipToAddress, setShipToAddress] = useState<AddressType | null>(
        defaultOrderContext.shipToAddress
    );
    const [signedTerms, setSignedTerms] = useState<boolean>(
        defaultOrderContext.signedTerms
    );
    const [userSignature, setUserSignature] = useState<string | null>(
        defaultOrderContext.userSignature
    );
    const [taxId, setTaxId] = useState<string>(defaultOrderContext.taxId);
    const [secondaryTaxId, setSecondaryTaxId] = useState<string>(
        defaultOrderContext.secondaryTaxId
    );
    const [paymentMethod, setPaymentMethod] = useState<PaymentMethod | null>(
        null
    );
    const [billingAddress, setBillingAddress] = useState<AddressType | null>(
        defaultOrderContext.billingAddress
    );
    const [signatureDateTimeStamp, setSignatureDateTimeStamp] =
        useState<Date | null>(defaultOrderContext.signatureDateTimeStamp);
    const [payPal, setPayPal] = useState<PayPalInfo | null>(
        defaultOrderContext.payPal
    );
    const [cashPayment, setCashPayment] = useState<CashPaymentInfo | null>(
        defaultOrderContext.cashPayment
    );
    const [bankWire, setBankWire] = useState<BankWireInfo | null>(
        defaultOrderContext.bankWire
    );
    const [isShipToOffice, setIsShipToOffice] = useState<boolean>(false);
    const [isCreditCardPaymentValid, setIsCreditCardPaymentValid] =
        useState<boolean>(defaultOrderContext.isCreditCardPaymentValid);
    const [selectedWillCallAddress, setSelectedWillCallAddress] = useState<
        string | null
    >(null);
    const [pickUpAddressKey, setPickUpAddressKey] = useState<string | null>(
        defaultOrderContext.pickUpAddressKey
    );
    const [tppPayload, setTppPayload] = useState<EventDetails | null>(null);

    const { isDigitalOnlyOrder } = useOrderCalculations(shoppingCart);

    const [feedbackSubmitted, setFeedbackSubmitted] = useState<boolean>(false);

    const [isEnrollmentCompleted, setIsEnrollmentCompleted] =
        useState<boolean>(false);

    const loadCachedOrderData = (): void => {
        const cachedData = sessionStorage.getItem("orderContext");
        if (cachedData) {
            const parsedData: OrderContextType = JSON.parse(cachedData);
            setOrderResult(parsedData.orderResult);
            setShippingMethodQuotes(parsedData.shippingMethodQuotes);
            setSelectedShippingMethod(parsedData.selectedShippingMethod);
            setSelectedPaymentMethod(parsedData.selectedPaymentMethod);
            setShipToAddress(parsedData.shipToAddress);
            setSignedTerms(parsedData.signedTerms);
            setUserSignature(parsedData.userSignature);
            setTaxId(parsedData.taxId);
            setSecondaryTaxId(parsedData.secondaryTaxId);
            setPaymentMethod(parsedData.paymentMethod);
            setBillingAddress(parsedData.billingAddress);
            setSignatureDateTimeStamp(parsedData.signatureDateTimeStamp);
            setPayPal(parsedData.payPal);
            setCashPayment(parsedData.cashPayment);
            setBankWire(parsedData.bankWire);
            setIsShipToOffice(parsedData.isShipToOffice);
            setIsCreditCardPaymentValid(parsedData.isCreditCardPaymentValid);
            setSelectedWillCallAddress(parsedData.selectedWillCallAddress);
            setPickUpAddressKey(parsedData.pickUpAddressKey);
            setTppPayload(parsedData.tppPayload);
            setFeedbackSubmitted(parsedData.feedbackSubmitted);
            setIsEnrollmentCompleted(parsedData.isEnrollmentCompleted);
        }
    };

    useEffect(() => {
        loadCachedOrderData();
    }, []);

    const updateCachedOrderData = useCallback((): void => {
        const dataToCache = {
            orderResult,
            shippingMethodQuotes,
            selectedShippingMethod,
            selectedPaymentMethod,
            shipToAddress,
            signedTerms,
            taxId,
            secondaryTaxId,
            paymentMethod,
            userSignature,
            billingAddress,
            signatureDateTimeStamp,
            payPal,
            cashPayment,
            bankWire,
            isShipToOffice,
            isCreditCardPaymentValid,
            selectedWillCallAddress,
            pickUpAddressKey,
            tppPayload,
            feedbackSubmitted,
            isEnrollmentCompleted
        };
        sessionStorage.setItem("orderContext", JSON.stringify(dataToCache));
    }, [
        orderResult,
        shippingMethodQuotes,
        selectedShippingMethod,
        selectedPaymentMethod,
        shipToAddress,
        signedTerms,
        taxId,
        secondaryTaxId,
        paymentMethod,
        userSignature,
        billingAddress,
        signatureDateTimeStamp,
        payPal,
        cashPayment,
        bankWire,
        isShipToOffice,
        isCreditCardPaymentValid,
        selectedWillCallAddress,
        pickUpAddressKey,
        tppPayload,
        feedbackSubmitted,
        isEnrollmentCompleted
    ]);

    useEffect(() => {
        updateCachedOrderData();
    }, [updateCachedOrderData]);

    const value = useMemo(
        () => ({
            orderResult,
            setOrderResult,
            shippingMethodQuotes,
            setShippingMethodQuotes,
            selectedShippingMethod,
            setSelectedShippingMethod,
            selectedPaymentMethod,
            setSelectedPaymentMethod,
            shipToAddress,
            setShipToAddress,
            signedTerms,
            setSignedTerms,
            taxId,
            setTaxId,
            secondaryTaxId,
            setSecondaryTaxId,
            paymentMethod,
            setPaymentMethod,
            userSignature,
            setUserSignature,
            billingAddress,
            setBillingAddress,
            signatureDateTimeStamp,
            setSignatureDateTimeStamp,
            payPal,
            setPayPal,
            cashPayment,
            setCashPayment,
            bankWire,
            setBankWire,
            isShipToOffice,
            setIsShipToOffice,
            isCreditCardPaymentValid,
            setIsCreditCardPaymentValid,
            selectedWillCallAddress,
            setSelectedWillCallAddress,
            pickUpAddressKey,
            setPickUpAddressKey,
            tppPayload,
            setTppPayload,
            feedbackSubmitted,
            setFeedbackSubmitted,
            isDigitalOnlyOrder,
            isEnrollmentCompleted,
            setIsEnrollmentCompleted
        }),
        [
            orderResult,
            shippingMethodQuotes,
            selectedShippingMethod,
            selectedPaymentMethod,
            shipToAddress,
            signedTerms,
            taxId,
            secondaryTaxId,
            paymentMethod,
            userSignature,
            billingAddress,
            signatureDateTimeStamp,
            payPal,
            cashPayment,
            bankWire,
            isShipToOffice,
            isCreditCardPaymentValid,
            selectedWillCallAddress,
            pickUpAddressKey,
            tppPayload,
            feedbackSubmitted,
            isDigitalOnlyOrder,
            isEnrollmentCompleted
        ]
    );

    const providerValue = useMemo(
        () => ({
            ...value,
            ...testValues
        }),
        [value, testValues]
    );

    return (
        <OrderContext.Provider value={providerValue}>
            {children}
        </OrderContext.Provider>
    );
}

export const useOrder = (): OrderContextType => useContext(OrderContext);
