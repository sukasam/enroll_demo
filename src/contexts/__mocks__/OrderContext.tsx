import { TransactionMethod } from "Services/Hydra/order/types";
import {
    AddressType,
    BankWireInfo,
    CashPaymentInfo,
    EventDetails,
    OrderResult,
    PayPalInfo,
    PaymentMethod,
    ShippingMethodQuoteType
} from "../types/OrderContextTypes";

export const useOrder = jest.fn().mockReturnValue({
    orderResult: null as OrderResult | null,
    setOrderResult: jest.fn(),
    shippingMethodQuotes: [] as ShippingMethodQuoteType[],
    setShippingMethodQuotes: jest.fn(),
    selectedShippingMethod: null as ShippingMethodQuoteType | null,
    setSelectedShippingMethod: jest.fn(),
    selectedPaymentMethod: null as TransactionMethod | null,
    setSelectedPaymentMethod: jest.fn(),
    shipToAddress: null as AddressType | null,
    setShipToAddress: jest.fn(),
    signedTerms: false,
    setSignedTerms: jest.fn(),
    userSignature: null as string | null,
    setUserSignature: jest.fn(),
    taxId: "",
    setTaxId: jest.fn(),
    secondaryTaxId: "",
    setSecondaryTaxId: jest.fn(),
    paymentMethod: null as PaymentMethod | null,
    setPaymentMethod: jest.fn(),
    billingAddress: null as AddressType | null,
    setBillingAddress: jest.fn(),
    signatureDateTimeStamp: null as Date | null,
    setSignatureDateTimeStamp: jest.fn(),
    payPal: null as PayPalInfo | null,
    setPayPal: jest.fn(),
    cashPayment: null as CashPaymentInfo | null,
    setCashPayment: jest.fn(),
    bankWire: null as BankWireInfo | null,
    setBankWire: jest.fn(),
    isShipToOffice: false,
    setIsShipToOffice: jest.fn(),
    isCreditCardPaymentValid: false,
    setIsCreditCardPaymentValid: jest.fn(),
    selectedWillCallAddress: null as string | null,
    setSelectedWillCallAddress: jest.fn(),
    pickUpAddressKey: null as string | null,
    setPickUpAddressKey: jest.fn(),
    tppPayload: null as EventDetails | null,
    setTppPayload: jest.fn(),
    isDigitalOnlyOrder: false,
    feedbackSubmitted: false,
    setFeedbackSubmitted: jest.fn(),
    isEnrollmentCompleted: false,
    setIsEnrollmentCompleted: jest.fn()
});

export default useOrder;
