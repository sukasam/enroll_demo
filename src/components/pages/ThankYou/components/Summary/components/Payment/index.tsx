/** @jsxImportSource @emotion/react */
import BankWireInfo from "Components/pages/Register/components/Payment/components/BankWire/components/BankWireInfo";
import LocalImage from "Components/shared/LocalImage";
import T, { useTranslate } from "Components/shared/Translate";
import getCardIcon from "Services/utils/getCardIcon";
import clsx from "clsx";
import { CreditCardType } from "credit-card-type/dist/types";
import Image from "next/image";
import { PaymentDetails } from "../../hooks";
import styles from "./styles";

interface PaymentProps {
    summaryExpanded: boolean;
    payment: PaymentDetails;
    cardInfo: CreditCardType[];
}

export default function Payment(props: PaymentProps): JSX.Element {
    const { summaryExpanded, payment, cardInfo } = props;
    const translate = useTranslate();

    const paymentMethods: {
        [key: string]: {
            iconSrc: string | (() => string);
            paymentMethodText: string | (() => string);
        };
    } = {
        CreditCard: {
            iconSrc: () => getCardIcon(payment.cardNumber),
            paymentMethodText: () => cardInfo[0]?.niceType
        },
        EWalletToken: {
            iconSrc: "/paypal.svg",
            paymentMethodText: translate("thank_you_paypal_account")
        },
        Cash: {
            iconSrc: "/cash.svg",
            paymentMethodText: translate("thank_you_cash_label")
        },
        BankWire: {
            iconSrc: "/bankwire.svg",
            paymentMethodText: translate("payment_bank_wire")
        }
    };

    let { iconSrc, paymentMethodText } = paymentMethods[payment.method];

    if (typeof iconSrc === "function") {
        iconSrc = iconSrc();
    }

    if (typeof paymentMethodText === "function") {
        paymentMethodText = paymentMethodText();
    }

    return (
        <section
            css={styles}
            className={clsx("payment", summaryExpanded || "mobile-hidden")}
        >
            <div className="payment-row" data-testid="payment_title">
                <T>thank_you_payment</T>
                {iconSrc && (
                    <Image
                        alt=""
                        height="24"
                        src={iconSrc}
                        width="32"
                        data-testid={`${payment.method.toLowerCase()}_image`}
                    />
                )}
                <span
                    className="name"
                    data-testid={`${payment.method.toLowerCase()}_name`}
                >
                    {paymentMethodText}
                </span>
            </div>
            {payment.method === "CreditCard" && (
                <div className="card-number" data-testid="credit_card_number">
                    {payment.cardNumber?.slice(-4)}
                </div>
            )}
            {payment.method === "BankWire" && (
                <div>
                    <div className="subtitle">
                        <T>thank_you_for_your_order_bankwire</T>
                    </div>

                    <div className="banner">
                        <LocalImage
                            className="icon"
                            height="18"
                            src="/svg/info.svg"
                            width="18"
                            priority={false}
                        />
                        <div
                            className="banner-header"
                            data-testid="bankwire_message"
                        >
                            <T>bankwire_message</T>
                        </div>
                    </div>
                </div>
            )}
            {payment.method === "Cash" && (
                <>
                    <div className="cash-details" data-testid="cash_name">
                        <strong>
                            <T>payment_cash_label</T>
                        </strong>
                    </div>
                    <div
                        className="message"
                        dangerouslySetInnerHTML={{
                            __html: translate("payment_cash_payment_message")
                        }}
                    />
                </>
            )}
            {payment.method === "BankWire" && (
                <div className="bankwire_container">
                    <BankWireInfo
                        titleKey="payment_beneficiary"
                        infoKey="payment_beneficiary_message"
                    />
                    <BankWireInfo
                        titleKey="payment_iban"
                        infoKey="payment_iban_number"
                    />
                    <BankWireInfo
                        titleKey="payment_bic"
                        infoKey="payment_bic_number"
                    />
                    <div className="order_instructions_container">
                        <strong>
                            <T>enter_id_message</T>
                        </strong>
                    </div>
                    <div>
                        <T>order_instructions</T>
                    </div>
                </div>
            )}
        </section>
    );
}
