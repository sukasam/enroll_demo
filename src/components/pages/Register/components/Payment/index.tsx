/** @jsxImportSource @emotion/react */
import PaymentMethodDetails from "Components/pages/Register/components/Payment/components/PaymentMethodDetails";
import PaymentMethodOptions from "Components/pages/Register/components/Payment/components/PaymentMethodOptions";
import ShippingMethodOptions from "Components/pages/Register/components/Payment/components/ShippingMethod";
import { useCountryConfig } from "Constants/countryConfig";
import { PaymentOptions } from "Constants/countryConfig/types";
import paymentOptions from "Constants/paymentOptions";
import { useOrder } from "Contexts/OrderContext";
import { useEffect, useMemo } from "react";
import styles from "./styles";

function PaymentSection(): JSX.Element {
    const { selectedPaymentMethod, setSelectedPaymentMethod } = useOrder();
    const config = useCountryConfig();

    const availablePaymentOptions = useMemo(
        () =>
            paymentOptions
                .map(option => {
                    const methodKey = `has${
                        option.transactionMethod.charAt(0).toUpperCase() +
                        option.transactionMethod.slice(1)
                    }` as keyof PaymentOptions;
                    return config?.marketExceptions?.paymentOptions?.[methodKey]
                        ? option
                        : null;
                })
                .filter(
                    (option): option is NonNullable<typeof option> =>
                        option !== null
                ),
        [config]
    );

    useEffect(() => {
        if (!selectedPaymentMethod && availablePaymentOptions.length > 0) {
            setSelectedPaymentMethod(
                availablePaymentOptions[0].transactionMethod
            );
        }
    }, [
        availablePaymentOptions,
        selectedPaymentMethod,
        setSelectedPaymentMethod
    ]);

    const filteredOptions = useMemo(
        () =>
            availablePaymentOptions.filter(
                option => option.transactionMethod !== selectedPaymentMethod
            ),
        [availablePaymentOptions, selectedPaymentMethod]
    );

    return (
        <div css={styles}>
            <div className="paymentSection">
                <PaymentMethodDetails
                    selectedPaymentMethod={selectedPaymentMethod}
                />
                <PaymentMethodOptions
                    onSelectMethod={setSelectedPaymentMethod}
                    paymentOptions={filteredOptions}
                />
            </div>
            <ShippingMethodOptions />
        </div>
    );
}

export default PaymentSection;
