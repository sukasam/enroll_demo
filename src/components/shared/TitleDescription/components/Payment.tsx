/** @jsxImportSource @emotion/react */
import T from "Components/shared/Translate";
import paymentOptions from "Constants/paymentOptions";
import { useOrder } from "Contexts/OrderContext";
import Image from "next/image";
import { useMemo } from "react";

export default function Payment(): JSX.Element | null {
    const { selectedPaymentMethod } = useOrder();

    const selectedPaymentOption = useMemo(
        () =>
            paymentOptions.find(
                paymentOption =>
                    paymentOption.transactionMethod === selectedPaymentMethod
            ),
        [selectedPaymentMethod]
    );

    if (!selectedPaymentOption) {
        return (
            <div className="description">
                <T>payment_select_payment_method</T>
            </div>
        );
    }

    return (
        <div className="description">
            <div className="icon">
                <Image
                    alt=""
                    height={18}
                    src={selectedPaymentOption.imageSrc}
                    width={24}
                    data-testid={`${selectedPaymentOption.testIdPrefix}_image`}
                />
            </div>
            <div
                className="text"
                data-testid={`${selectedPaymentOption.testIdPrefix}_label`}
            >
                <T>{selectedPaymentOption.optionLabel}</T>
            </div>
        </div>
    );
}
