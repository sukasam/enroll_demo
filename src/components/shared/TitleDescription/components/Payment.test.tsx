import { render } from "@testing-library/react";
import { useOrder } from "Contexts/OrderContext";
import Payment from "./Payment";

// Mock the necessary hooks and components
jest.mock("Contexts/OrderContext", () => ({
    useOrder: jest.fn()
}));

jest.mock("Constants/paymentOptions", () => [
    {
        transactionMethod: "CreditCard",
        imageSrc: "/images/credit-card.png",
        optionLabel: "Credit Card",
        testIdPrefix: "credit_card"
    },
    {
        transactionMethod: "PayPal",
        imageSrc: "/images/paypal.png",
        optionLabel: "PayPal",
        testIdPrefix: "paypal"
    }
]);

jest.mock(
    "Components/shared/Translate",
    () =>
        function Translate({
            children
        }: {
            children: React.ReactNode;
        }): JSX.Element {
            return <span>{children}</span>;
        }
);

jest.mock(
    "next/image",
    () =>
        function Image(props: any): JSX.Element {
            return <img {...props} alt="" />;
        }
);

describe("Payment", () => {
    beforeEach(() => {
        (useOrder as jest.Mock).mockReturnValue({
            selectedPaymentMethod: null
        });
    });

    it("renders correctly when no payment method is selected", () => {
        const { asFragment } = render(<Payment />);
        expect(asFragment()).toMatchSnapshot();
    });

    it("renders correctly when CreditCard payment method is selected", () => {
        (useOrder as jest.Mock).mockReturnValue({
            selectedPaymentMethod: "CreditCard"
        });

        const { asFragment } = render(<Payment />);
        expect(asFragment()).toMatchSnapshot();
    });

    it("renders correctly when PayPal payment method is selected", () => {
        (useOrder as jest.Mock).mockReturnValue({
            selectedPaymentMethod: "PayPal"
        });

        const { asFragment } = render(<Payment />);
        expect(asFragment()).toMatchSnapshot();
    });
});
