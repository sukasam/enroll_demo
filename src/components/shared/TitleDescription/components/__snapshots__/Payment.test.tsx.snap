// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Payment renders correctly when CreditCard payment method is selected 1`] = `
<DocumentFragment>
  <div
    class="description"
  >
    <div
      class="icon"
    >
      <img
        alt=""
        data-testid="credit_card_image"
        height="18"
        src="/images/credit-card.png"
        width="24"
      />
    </div>
    <div
      class="text"
      data-testid="credit_card_label"
    >
      <span>
        Credit Card
      </span>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Payment renders correctly when PayPal payment method is selected 1`] = `
<DocumentFragment>
  <div
    class="description"
  >
    <div
      class="icon"
    >
      <img
        alt=""
        data-testid="paypal_image"
        height="18"
        src="/images/paypal.png"
        width="24"
      />
    </div>
    <div
      class="text"
      data-testid="paypal_label"
    >
      <span>
        PayPal
      </span>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Payment renders correctly when no payment method is selected 1`] = `
<DocumentFragment>
  <div
    class="description"
  >
    <span>
      payment_select_payment_method
    </span>
  </div>
</DocumentFragment>
`;
